DataFormats/Detectors/TRD/include/DataFormatsTRD/TrapConfig3.h

```diff
@@ -0,0 +1,619 @@
+
+  // return all the mcm for a particular register address
+  void getAllMCMByAddress(int registeraddress, std::array<uint32_t, o2::trd::constants::MAXMCMCOUNT>& mcms);
+
+  // return all the config data in an unpacked array.
+  void getAll(std::array<uint32_t, kLastReg * o2::trd::constants::MAXMCMCOUNT>& configdata);
+
+  // return the full unpacked config event.
+  bool printRegister(int addr, int det, int rob, int mcm);
+  bool printRegister(TrapRegInfo* reg, int det, int rob, int mcm);
+
+  const std::array<o2::trd::TrapRegInfo, kLastReg>& getTrapRegisters() { return mTrapRegisters; }
+
+  // population pending
+  void setConfigVersion(uint32_t version) { mTrapConfigVersion = version; }           // these must some how be gotten from git or wingdb.
+  void setConfigNumber(uint32_t number) { mTrapConfigNumber = number; }               // these must be gotten from git or wingdb.
+  void setConfigSavedVersion(uint16_t version) { mTrapConfigSavedVersion = version; } // the version that is saved, for the ability to later save the config differently.
+  uint32_t getConfigVersion() { return mTrapConfigVersion; }                          // these must some how be gotten from git or wingdb.
+  uint32_t getConfigName() { return mTrapConfigNumber; }                              // these must be gotten from git or wingdb.
+  uint16_t getConfigSavedVersion() { return mTrapConfigSavedVersion; }                // the version that is saved, for the ability to later save the config differently.
+
+  inline bool operator==(const TrapConfig3& other);
+
+  // for compliance with the same interface to o2::trd::TrapConfig and its run1/2 inherited interface:
+  // only these 2 are used in the simulations.
+  uint32_t getDmemUnsigned(uint32_t address, int detector, int rob, int mcm);
+  uint32_t getTrapReg(uint32_t index, int detector, int rob, int mcm);
+
+ private:
+  std::array<o2::trd::TrapRegInfo, kLastReg> mTrapRegisters;                              //!< store of layout of each block of mTrapRegisterSize, populated via initialiseRegisters
+  std::bitset<o2::trd::constants::MAXMCMCOUNT> mMCMDataHasChanged;                        //!< does the mcm actually contain data.
+  std::array<uint32_t, kTrapRegistersSize * o2::trd::constants::MAXMCMCOUNT> mConfigData; //!< one block of data per mcm.
+  std::map<uint16_t, uint16_t> mTrapRegistersAddressIndexMap;                             //!< map of address into mTrapRegisters, populated at the end of initialiseRegisters
+  void initialiseRegisters();
+
+  uint32_t mTrapConfigNumber;       //!< the version number coming from the config that is written to the traps, from config register C09CPU01
+  uint32_t mTrapConfigVersion;      //!< the version of the coming from the config that is written to the traps, from DigitHCHeader (svn version number)
+  uint16_t mTrapConfigSavedVersion; //!< the version that is saved, for the ability to later save the config differently.
```

Comment:
I might be misremembering this, but wasnt there talk to convert the svn repo to git?

Reply:
That is a whole nother discussion. This is just reflecting the current state of the outside code.